src/runtime/str_impl.rs

```diff
@@ -1319,6 +1397,57 @@ fn process_match(matched: &[u8], subst: &[u8], w: &mut impl Write) -> io::Result
     Ok(())
 }
 
+/// Helper function for `subst_gen` function; handles the syntax for &, \0, \1, etc...
+fn process_match_gen(matched: Captures, subst: &[u8], w: &mut impl Write) -> io::Result<()> {
+    let mut start = 0;
+    let mut escaped = false;
+    for (i, b) in subst.iter().cloned().enumerate() {
+        match b {
+            b'0'..=b'9' => {
```

Comment:
Does this mean we only handle single-digit matches? Would it be better to handle an arbitrary integer?

Reply:
Well, this is what gawk does...

![image](https://user-images.githubusercontent.com/10363282/163771975-2caa6ac5-4e26-433f-9e4b-1d8454e9c0e7.png)

I kinda don't like the way they did it too, but I think it's better to lean on compat here

Reply:
Ah, you are correct! Thanks for the reference. I agree with keeping gawk's behavior here; I think if we ever wanted to break the semantics we'd just add a different function.